File: backend/src/Models/Personagem.cs
using System.ComponentModel.DataAnnotations;


namespace backend.Models;

public record Personagem
{
    public long PersonagemId { get; set; }

    public string? ImageUrl { get; set; }
    
    public string? Nome { get; set; }

    public int Pv { get; set; }

    public int PvTotal { get; set; }

    public int Armadura { get; set; }

    public int DadoDano { get; set; }

    public int Nivel { get; set; }

    public string? Classe { get; set; }

    public string? DescricaoUm { get; set; }

    public string? Equipamento { get; set; }

    public string? DescricaoDois { get; set; }
    
    public Habilidade? Habilidade { get; set; }

    public override string ToString()
    {
        return $"""
                Id = {PersonagemId.ToString()}
                Nome = {Nome ?? ""}
                Pv = {Pv.ToString()}
                Armadura = {Armadura.ToString()}
                DadoDano = {DadoDano.ToString()}
                Nivel = {Nivel.ToString()}
                """;
    }
}

File: frontend/src/Models/Personagem.ts
export interface Habilidade {
    id: number;
    forca: number;
    destreza: number;
    constituicao: number;
    inteligencia: number;
    sabedoria: number;
    carisma: number;
}

// Interface para Personagem.ts
export interface Personagem {
    personagemId: number;
    imageUrl: string;
    nome: string;
    pv: number;
    pvTotal: number;
    armadura: number;
    dadoDano: number;
    nivel: number;
    classe: string;
    descricaoUm: string;
    equipamento: string;
    descricaoDois: string;
    habilidade: Habilidade;
}

export interface FetchPersonagemList {
    status: boolean;
    message: string;
    data: Personagem[];
}
export interface FetchPersonagem {
    status: boolean;
    message: string;
    data: Personagem;
}